pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2318,4 +2318,9 @@ self: super: {
   # Too strict bounds on chell: https://github.com/fpco/haskell-filesystem/issues/24
   system-fileio = doJailbreak super.system-fileio;
 
+  # 2022-02-28: Unmaintained. Distribution.Simple must have changed since 2016
+  digits = appendPatches [
+    ./patches/digits-fix-162266.patch
+  ] super.digits;
```

Comment:
In general we try to not carry around patches in Nixpkgs if at all possible.

Is there some upstream you can send this patch to, and then pull it in here with `fetchpatch`?  There should be a bunch of examples of `fetchpatch` in this file.  Also make sure to post a link so that other maintainers can see if it ever gets merged in.

If there is no upstream, then maybe you could take over maintenance on Hackage?

Reply:
Upstream doesn't exist anymore: https://bitbucket.org/sffubs/digits and https://hackage.haskell.org/package/digits

> If there is no upstream, then maybe you could take over maintenance on Hackage?

I don't really know/use Haskell. I just want to package a Haskell app.

Reply:
@bbigras What app are you trying to package?  Have you talked to those app maintainers about this dependency?

In general, we try to get Haskell-related stuff fixed upstream, since it benefits the entire Haskell community (instead of just selfishly benefiting only the Nix community).

If you don't want to put in the effort to get this fixed upstream, then we have to decide whether we want to put in the effort to bloat nixpkgs with patch files, carry around these nixpkgs-specific changes, etc.  The decision would be swayed by how popular the app is and how many people would potentially use it.

Reply:
I wanted to package #162211. But this is clearly a wasted of my time after I already spent hours on this.

Reply:
Upstream seems to be a pain to package indeed, as it uses Stackage LTS 16 and even pins some packages to git revisions. Getting upstream to update / pull out problematic dependencies would help a lot, I'd guess.